src/aihwkit/nn/modules/linear_sliced.py

```diff
@@ -0,0 +1,197 @@
+        tile_module_class: Optional[Type] = None,
+        significance_factors: Optional[List[int]] = None,
+    ):
+        # Call super()
+        Linear.__init__(self, in_features, out_features, bias=bias)
+
+        # Create tile
+        if rpu_config is None:
+            # pylint: disable=import-outside-toplevel
+            from aihwkit.simulator.configs.configs import SingleRPUConfig
+
+            rpu_config = SingleRPUConfig()
+
+        AnalogLayerBase.__init__(self)
+
+        if tile_module_class is None:
+            tile_module_class = rpu_config.get_default_tile_module_class(out_features, in_features)
+
+        self.significance_factors = []
+        if significance_factors is None:
+            if evenly_sliced is True:
+                self.significance_factors = [1] * number_slices
+            else:
+                # We increase the significance by 2 for every slice
+                self.significance_factors = [1]
+                for _ in range(1, number_slices):
+                    self.significance_factors.append(self.significance_factors[-1]*2)
+        else:
+            if len(significance_factors) != number_slices:
+                raise ModuleError("Length of factors must equal number of slices exactly")
+            self.significance_factors = significance_factors
+
+        self.analog_slices = ParameterList(AnalogLinear(in_features, out_features, 
+                                                        bias, rpu_config, 
+                                                        tile_module_class) for i in range(number_slices))
+        # Unregister weight/bias as a parameter.
+        self.unregister_parameter("weight")
+        if bias:
+            self.unregister_parameter("bias")
+        else:
+            # Seems to be a torch bug.
+            self._parameters.pop("bias", None)
+        self.bias = bias
+
+        self.reset_parameters()
+        
+
+    def reset_parameters(self) -> None:
+        """Reset the parameters (weight and bias)."""
+        if hasattr(self, "analog_module"):
+            for weight_slice in self.analog_slices:
+                bias = weight_slice.bias
+                weight_slice.weight, weight_slice.bias = weight_slice.get_weights()  # type: ignore
+                weight_slice.super().reset_parameters()
+                weight_slice.set_weights(weight_slice.weight, weight_slice.bias)  # type: ignore
+                weight_slice.weight, weight_slice.bias = None, bias
+
+    def forward(self, x_input: Tensor) -> Tensor:
+        """Compute the forward pass."""
+        # pylint: disable=arguments-differ, arguments-renamed
+        forward_output = 0 #to initialize the forward_output for the loop
+        for idx, weight_slice in enumerate(self.analog_slices):
+            forward_output += (weight_slice(x_input) * self.significance_factors[idx]) 
+
+        return forward_output  # type: ignore
+
+    @classmethod
+    def from_digital(
```

Comment:
It would be better to add some arguments in this from_digital call to set the slices etc. To use the convert_to_digital one has to use a custom mapping anyway. So the best would be to add a `convert_to_digital_sliced` function that converts all `Linear` to AnalogLinearSliced` which also accepts and passes the slicing parameters during the construction.

Reply:
convert_to_analog_sliced, you mean?

Reply:
Right